omero/developers/Java.rst

```diff
@@ -990,6 +990,64 @@ in OMERO. Similar approach can be applied when uploading an image.
         store.close();
     }
 
+Sudo (working within another user's context)
+--------------------------------------------
+
+The next code snippet shows how you can work within another user's context. This
+could for example be a data analyst doing some analysis on behalf of a user and
+attaching the results to the user's data. The important point is that the user will
+be the owner of these results and can work with them as usual. The user and 'analyst'
+do not have to be member of a read-annotate group (see :doc:`Server/Permissions`),
+but the 'analyst' has to be a 'light administrator' with 'sudo' permission,
+see :doc:`Server/LightAdmins`.
+
+::
+
+    AdminFacility admin = gateway.getFacility(AdminFacility.class);
+
+    // Look up the experimenter to sudo for
+    ExperimenterData sudoUser = admin.lookupExperimenter(ctx, sudoUserName);
+
+    // Create a SecurityContext for this user within the user's default group
+    // and set the 'sudo' flag (i.e. all operations using this context will
+    // be performed as this user)
+    SecurityContext sudoCtx = new SecurityContext(sudoUser.getGroupId());
+    sudoCtx.setExperimenter(sudoUser);
+    sudoCtx.sudo();
+
+    // Get the sudouser's datasets
+    BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
+    Collection<DatasetData> datasets = browse.getDatasets(sudoCtx, sudoUser.getId());
+    Iterator<DatasetData> i = datasets.iterator();
+    DatasetData first = null;
```

Comment:
maybe `DatasetData first = datasets.iterator().next()` would suffice to illustrate the use of the first dataset. I realize there is a potential for an exception but that happens with the code as is if the user has no dataset and I think that's ok because the code is there for illustration.

Reply:
Could simply add a brief comment that makes the assumption explicit.

Reply:
that's true